test.sh

```diff
@@ -21,48 +25,135 @@ function prepareOutputDir() {
 }
 
+function printProxyConfiguration() {
+  echo '-----------------------test.sh------------------------------'
+  echo "Print Nginx proxy server configuration"
+  echo '------------------------------------------------------------'
+  echo 'Note: in order to change configuration, you need to edit the file: default.conf '
+
+  # get conjur client container id
+  conjur_proxy_cid=$(docker-compose ps -q conjur-proxy-nginx)
+  exec_command='nginx-debug -T'
+  docker exec ${conjur_proxy_cid} ${exec_command}
+
+}
+
+
+function initializeCert() {
+  echo '-----------------------test.sh------------------------------'
+  echo "Fetch certificate using client cli"
+  echo '------------------------------------------------------------'
+
+  # get conjur client container id
+  conjur_client_cid=$(docker-compose ps -q client)
+
+  # get the pem file from conjur server
+  CONJUR_ACCOUNT="cucumber"
+  CONJUR_PROXY="https://conjur-proxy-nginx"
+
+  echo "remove old pem file"
+  rm -rf /test-cert/*
+
+  echo "fetch pem file from proxy https server"
+  exec_command='echo yes | conjur init -u https://conjur-proxy-nginx -a cucumber > tmp.out 2>&1'
+  docker exec ${conjur_client_cid} /bin/bash -c "$exec_command"
+
+  echo "print command output"
+  print_command="cat tmp.out"
+  docker exec ${conjur_client_cid} ${print_command}
+
+  echo "convert pem to der file and copy it to share memory"
+  convert_command="openssl x509 -outform der -in /root/conjur-cucumber.pem -out /test-cert/conjur-cucumber.der"
+  docker exec ${conjur_client_cid} ${convert_command}
+
+  echo "import cert inside test https container"
+
+  # get conjur test https container id
+  conjur_test_cid=$(docker-compose ps -q test-https)
+
+  JRE_HOME='/usr/lib/jvm/java-7-openjdk-amd64/jre'
+  import_command="keytool -import -alias cucumber -v -trustcacerts -noprompt \
+     -keystore $JRE_HOME/lib/security/cacerts \
+     -file /test-cert/conjur-cucumber.der -storepass changeit"
+  docker exec ${conjur_test_cid} ${import_command}
+
+
+}
+
 function runTests() {
+  echo '-----------------------test.sh------------------------------'
   echo "Running tests"
-  echo '-----'
+  echo '------------------------------------------------------------'
 
-  api_key=$(docker-compose exec -T possum rails r "print Credentials['cucumber:user:admin'].api_key")
+  api_key_admin=$(docker-compose exec -T conjur rails r "print Credentials['cucumber:user:admin'].api_key")
 
   # Execute tests
   docker-compose run --rm \
-    -e CONJUR_CREDENTIALS="admin:$api_key" \
+    -e CONJUR_CREDENTIALS="admin:$api_key_admin" \
     test \
-    bash -c "mvn test"
+      bash -c "mvn test"
+
+
+
+}
+
+function runHTTPSTests() {
+  echo '-----------------------test.sh------------------------------'
+  echo "Running https tests"
+  echo '------------------------------------------------------------'
+
+  api_key_admin=$(docker-compose exec -T conjur rails r "print Credentials['cucumber:user:admin'].api_key")
```

Comment:
These API keys should all be output when we load the policy, why are we grabbing them from rails console?

Reply:
Because I need the rails console to grab the admin api key in order to load the policy, so I used the same logic also when I needed the user $ host api keys to run the tests.
Does its matter if I bring those keys from the policy output or from the rails command?

Reply:
i'm not sure, and i think this is fine for merge. We can clean up in future commits before release, if need be.